source/Conversation.h

```diff
@@ -72,17 +72,61 @@ class Conversation {
 	// the user to select; others just automatically continue to another node.
 	// Nodes may also display images or include conditional branches.
 	bool IsChoice(int node) const;
+	// If the given node is a choice node, check how many choices it offers.
 	int Choices(int node) const;
+	// Check if the given conversation node is a conditional branch.
 	bool IsBranch(int node) const;
+	// Check if the given conversation node performs an action.
 	bool IsAction(int node) const;
-	const ConditionSet &Branch(int node) const;
+	const ConditionSet &Conditions(int node) const;
 	const GameAction &GetAction(int node) const;
-	const std::string &Text(int node, int choice = 0) const;
+	const std::string &Text(int node, int element = 0) const;
 	const Sprite *Scene(int node) const;
-	int NextNode(int node, int choice = 0) const;
+	int NextNode(int node, int element = 0) const;
+	// Returns whether the given node should be displayed.
+	// Returns false if:
+	// - The node (or element) is out of range.
+	// - The node is not a choice node, and element is non-zero.
+	// - The node (or element) has conditions and those conditions are not met.
+	// and true otherwise.
+	bool ShouldShowText(const std::map<std::string, int64_t> &vars, int node, int element = 0) const;
+	// Returns true if the given node index is in the range of valid nodes for
+	// this Conversation.
+	// Note: The "outcomes" listed above do *not* refer to nodes *within* the
+	// Conversation, and are therefore considered invalid by this function!
```

Comment:
What about:

> Note: This function only considers actual Conversation nodes to be valid choices. The negative "outcome" values (listed above) are special sentinel values, rather than indices of Conversation nodes, and are therefore considered invalid by this function.